File: app/src/main/java/com/zamzzz/killwifi/SplashActivity.java
package com.zamzzz.killwifi;

import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.animation.LinearInterpolator;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class SplashActivity extends AppCompatActivity {

    private ImageView ivSkull;
    private TextView tvTitle, tvSubtitle, tvLoading, tvVersion;
    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        // INIT VIEWS
        ivSkull = findViewById(R.id.iv_skull);
        tvTitle = findViewById(R.id.tv_title);
        tvSubtitle = findViewById(R.id.tv_subtitle);
        tvLoading = findViewById(R.id.tv_loading);
        tvVersion = findViewById(R.id.tv_version);
        progressBar = findViewById(R.id.progress_bar);

        // ANIMASI FADE IN TITLE
        Animation fadeIn = AnimationUtils.loadAnimation(this, android.R.anim.fade_in);
        fadeIn.setDuration(1500);
        tvTitle.startAnimation(fadeIn);
        tvSubtitle.startAnimation(fadeIn);

        // ANIMASI PULSE SKULL
        ObjectAnimator pulseX = ObjectAnimator.ofFloat(ivSkull, "scaleX", 1f, 1.15f, 1f);
        ObjectAnimator pulseY = ObjectAnimator.ofFloat(ivSkull, "scaleY", 1f, 1.15f, 1f);
        pulseX.setDuration(1200);
        pulseY.setDuration(1200);
        pulseX.setRepeatCount(ValueAnimator.INFINITE);
        pulseY.setRepeatCount(ValueAnimator.INFINITE);
        pulseX.setInterpolator(new LinearInterpolator());
        pulseY.setInterpolator(new LinearInterpolator());
        pulseX.start();
        pulseY.start();

        // ANIMASI ROTATE GLOW
        ObjectAnimator rotate = ObjectAnimator.ofFloat(ivSkull, "rotation", 0f, 360f);
        rotate.setDuration(8000);
        rotate.setRepeatCount(ValueAnimator.INFINITE);
        rotate.setInterpolator(new LinearInterpolator());
        rotate.start();

        // ANIMASI PROGRESS BAR
        ObjectAnimator progressAnim = ObjectAnimator.ofInt(progressBar, "progress", 0, 100);
        progressAnim.setDuration(2500);
        progressAnim.start();

        // ANIMASI LOADING TEXT BLINK
        ObjectAnimator blink = ObjectAnimator.ofFloat(tvLoading, "alpha", 1f, 0.3f, 1f);
        blink.setDuration(800);
        blink.setRepeatCount(ValueAnimator.INFINITE);
        blink.start();

        // PINDAH KE MAIN ACTIVITY SETELAH 2.8 DETIK
        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
                finish();
            }
        }, 2800);
    }

    @Override
    public void onBackPressed() {
        // DISABLE BACK BUTTON DI SPLASH
    }
}

File: app/src/main/java/com/tycept/chatapp/SplashActivity.kt
package com.tycept.chatapp

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.animation.AccelerateDecelerateInterpolator
import android.view.animation.OvershootInterpolator
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val logo = findViewById<ImageView>(R.id.splashLogo)
        val title = findViewById<TextView>(R.id.splashTitle)
        val subtitle = findViewById<TextView>(R.id.splashSubtitle)

        logo.scaleX = 0f
        logo.scaleY = 0f
        logo.alpha = 0f
        title.alpha = 0f
        title.translationY = 40f
        subtitle.alpha = 0f

        val logoScaleX = ObjectAnimator.ofFloat(logo, "scaleX", 0f, 1.15f, 1f)
        val logoScaleY = ObjectAnimator.ofFloat(logo, "scaleY", 0f, 1.15f, 1f)
        val logoAlpha = ObjectAnimator.ofFloat(logo, "alpha", 0f, 1f)

        val logoSet = AnimatorSet()
        logoSet.playTogether(logoScaleX, logoScaleY, logoAlpha)
        logoSet.duration = 650
        logoSet.interpolator = OvershootInterpolator(1.6f)

        val titleAlpha = ObjectAnimator.ofFloat(title, "alpha", 0f, 1f)
        val titleTrans = ObjectAnimator.ofFloat(title, "translationY", 40f, 0f)
        val titleSet = AnimatorSet()
        titleSet.playTogether(titleAlpha, titleTrans)
        titleSet.duration = 450
        titleSet.interpolator = AccelerateDecelerateInterpolator()
        titleSet.startDelay = 350

        val subtitleAlpha = ObjectAnimator.ofFloat(subtitle, "alpha", 0f, 1f)
        subtitleAlpha.duration = 400
        subtitleAlpha.startDelay = 550

        val fullSet = AnimatorSet()
        fullSet.playTogether(logoSet, titleSet, subtitleAlpha)
        fullSet.start()

        Handler(Looper.getMainLooper()).postDelayed({
            startActivity(Intent(this, ChatListActivity::class.java))
            overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
            finish()
        }, 1900)
    }
}
